backend/src/main/java/gov/cdc/usds/simplereport/service/PatientLinkService.java

```diff
@@ -33,21 +32,41 @@ public PatientLink getPatientLink(UUID internalId) {
             () -> new IllegalGraphqlArgumentException("No patient link with that ID was found"));
   }
 
+  public PatientLink getRefreshedPatientLink(UUID internalId) {
+    PatientLink pl =
+        plrepo
+            .findById(internalId)
+            .orElseThrow(
+                () ->
+                    new IllegalGraphqlArgumentException("No patient link with that ID was found"));
+    pl.refresh();
+    return plrepo.save(pl);
+  }
+
   public boolean verifyPatientLink(UUID internalId, LocalDate birthDate)
       throws ExpiredPatientLinkException {
     try {
       PatientLink patientLink = getPatientLink(internalId);
+      if (patientLink.isLockedOut()) {
+        throw new ExpiredPatientLinkException();
+      }
+
       TestOrder testOrder = patientLink.getTestOrder();
       Person patient = testOrder.getPatient();
+
       if (testOrder.getPatient().getBirthDate().equals(birthDate)) {
         if (patientLink.isExpired()) {
           throw new ExpiredPatientLinkException();
         }
         contextHolder.setContext(patientLink, testOrder, patient);
         return true;
```

Comment:
With non-transient accounts, you would want to reset the failed attempt counter to 0 here, but I'm not sure that's necessary for patient accounts.

Reply:
Ah, yeah. It can't hurt, so I'll make that adjustment!